### DISCUSSION TITLE
How to add Multiple Markers

### DISCUSSION THREAD

**Discussion by jepongski** [osmandapp/OsmAnd#11589] [General]
How can we add multiple markers in OSMand using intent addMapMarker, When i put iteration, it creates multiple OSMAnd window/Activity. and when I am trying to use the AIDL addMapMarker but does not work and i am getting a toast message "OSM and" service NOT bind.

**Comment by vshcherb** [comment]
Did you try example https://github.com/osmandapp/osmand-api-demo/tree/master/OsmAnd-api-sample ? https://download.osmand.net/latest-night-build/OsmAnd-api-sample.apk. I think it has a working example with AIDL

**Reply by jepongski** [reply]
I tried the sample demo API and it seems that AIDL part is not working. Intent is working fine. the AIDL helper returns "OSMAnd" Service Not Bind

**Reply by Chumva** [reply]
What OsmAnd version you tested with demo API? Try to test API demo with net.osmand.plus.

This is an example for binding osmand aidl service - 
https://github.com/osmandapp/osmand-api-demo/blob/e232fb16553ce4cf4936bb7fdadb753c4e6d5e8f/OsmAnd-api-sample/app/src/main/java/net/osmand/osmandapidemo/OsmAndAidlHelper.java#L279

**Reply by vshcherb** [reply]
Demo is connected only to OsmAnd+, so you need to change the code to connect to 'net.osmand' instead

**Reply by jepongski** [reply]
Thanks for your reply. I am able to use the AIDL now. my next question is, how will I add the OSMandAIDLhelper.java to other project. I am not able to implement the dependency "com.github.osmandapp:Osmand:master-SNAPSHOT" it throws an error that it could not be found.

**Reply by vshcherb** [reply]
It takes some time to be built, try again and it will be loaded from ```maven {
			url "https://jitpack.io"
		}```
